src/main/java/core/controller/MovieSessionController.java

```diff
@@ -0,0 +1,67 @@
+package core.controller;
+
+import core.model.MovieSession;
+import core.model.dto.MovieSessionRequestDto;
+import core.model.dto.MovieSessionResponseDto;
+import core.service.MovieSessionService;
+import core.util.mapper.MovieSessionMapper;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequestMapping("/movie-sessions")
+@RestController
+public class MovieSessionController {
+    private final MovieSessionService movieSessionService;
+    private final MovieSessionMapper movieSessionMapper;
+    
+    public MovieSessionController(MovieSessionService movieSessionService,
+                                  MovieSessionMapper movieSessionMapper) {
+        this.movieSessionService = movieSessionService;
+        this.movieSessionMapper = movieSessionMapper;
+    }
+    
+    @GetMapping("/{id}")
+    public MovieSessionResponseDto get(@PathVariable Long id) {
+        return movieSessionMapper.mapToDto(movieSessionService.get(id));
+    }
+    
+    @GetMapping("/available")
+    public List<MovieSessionResponseDto> getAvailable(@RequestParam(name = "movie-id") Long movieId,
```

Comment:
we usually write parameters in camelCase, so we can remove `name` attribute

Reply:
but in REST specification it was dashed? I don't mind changing it, but just in case

Reply:
in the URL itself we use dashes, but it request parameters we can use camelCase